config/nrfconnect/chip-module/generate_factory_data.cmake

```diff
@@ -0,0 +1,225 @@
+file(GENERATE
+    OUTPUT ${output_path}/${factory_data_target}_cbor.args
+    CONTENT ${cbor_script_args})
+
+# execute second script to create a hex file containing factory data in cbor format
+separate_arguments(separated_cbor_script_args NATIVE_COMMAND ${cbor_script_args})
+set(factory_data_hex ${output_path}/${factory_data_target}.hex)
+# return output hex to parent scope
+set(${output_hex} ${factory_data_hex} PARENT_SCOPE)
+add_custom_command(OUTPUT ${factory_data_hex}
+    COMMAND ${Python3_EXECUTABLE} ${script_path} ${separated_cbor_script_args}
+    COMMENT "Generating factory data HEX file..."
+    DEPENDS ${factory_data_target}
+    )
+
+endfunction()
+
+# Generate factory data partition using given args
+#
+# 
+# During generation process a some file will be created in zephyr's build directory:
+# - merged_fd.hex a file containing firmware and factory data merged to single file
+# - factory_data.hex a file containing only a factory data partition including proper offset
+#
+# [Args]:
+#   target_name         - a name for target to generate factory_data
+function(nrfconnect_generate_factory_data target_name)
+
+find_package(Python REQUIRED)
+
+# CHIP_ROOT must be provided as a reference set all localization of scripts
+if(NOT CHIP_ROOT)
+message(FATAL_ERROR "CHIP_ROOT variable is not set, please add it to CMakeLists.txt file")
+endif()
+
+# Localize all scripts needed to generate factory data partition
+set(FACTORY_DATA_SCRIPT_PATH ${CHIP_ROOT}/scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py)
+set(GENERATE_CBOR_SCRIPT_PATH ${CHIP_ROOT}/scripts/tools/nrfconnect/nrfconnect_generate_partition.py)
+SET(MERGE_HEX_SCRIPT_PATH ${CHIP_ROOT}/config/nrfconnect/chip-module/merge_factory_data.py)
+set(FACTORY_DATA_SCHEMA_PATH ${CHIP_ROOT}/scripts/tools/nrfconnect/nrfconnect_factory_data.schema)
+set(OUTPUT_FILE_PATH ${APPLICATION_BINARY_DIR}/zephyr)
+
+# check if factory data has been already generated
+if(EXISTS ${OUTPUT_FILE_PATH}/${target_name}.hex)
+    message("\nFactory data was created previously, to generate a new one delete ${target_name}.hex file in your build directory\n")
+else()
+# create a JSON file with all factory data
+nrfconnect_create_factory_data_json(${target_name} 
+                                    ${FACTORY_DATA_SCRIPT_PATH} 
+                                    ${FACTORY_DATA_SCHEMA_PATH} 
+                                    ${OUTPUT_FILE_PATH})
+
+# create a .hex file with factory data in CBOR format based on the JSON file created previously 
+nrfconnect_create_factory_data_hex_file(${target_name} 
+                                        ${GENERATE_CBOR_SCRIPT_PATH} 
+                                        ${OUTPUT_FILE_PATH} 
+                                        ${target_name}_hex)
+
+endif()
+
+if(CONFIG_CHIP_MERGE_FACTORY_DATA_WITH_FIRMWARE)                                   
+    # set custom target for merging factory_data hex file
+    add_custom_target(${target_name}_merge
```

Comment:
Do we need another target? Can't we set `${target_name}.hex` directly as the _PM_TARGET?

Reply:
I can see CMake script doesn't work without it. In the same way, it is done in e.g. in mcuboot.
`ninja: error: 'zephyr/factory_data.hex', needed by 'zephyr/merged.hex', missing and no known rule to make it`